commons/src/main/java/com/powsybl/commons/exceptions/UncheckedClassCastExceptionException.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.exceptions;
+
+/**
+ *
+ * @author Ghiles Abdellah <ghiles.abdellah at rte-france.com>
+ */
+public class UncheckedClassCastExceptionException extends RuntimeException {
+
+    private static final long serialVersionUID = 154490417885926671L;
+
+    public UncheckedClassCastExceptionException(ClassCastException cause) {
```

Comment:
You should probably add a default constructor or a constructor that takes a message as parameter to avoid explicit instanciation of a ClassCastException. For instance, in your case, you will have to write:
```
if (! X instanceOf Y) {
    throw new UncheckedClassCastException(new ClassCastException(“X of type Z cannot be converted to Y“));
}

Reply:
added a new message constructor in the new version